contracts/paratii/TcrPlaceholder.sol

```diff
@@ -0,0 +1,169 @@
+pragma solidity ^0.4.18;
+
+import "./ParatiiToken.sol";
+
+/// WARNING , NOT PRODUCTION READY, based on ADCHAIN Registry.
+/// THIS IS JUST A PLACEHOLDER..
+contract TcrPlaceholder {
+
+  // Events
+  // ---------
+
+  event _Application(string videoId, uint deposit);
+  event _NewVideoWhitelisted(string videoId);
+
+  struct Listing {
+    uint applicationExpiry; // Expiration date of apply stage
+    bool whitelisted;       // registration status
+    address owner;          // Owner of the video.
+    uint unstakedDeposit;   // unlocked tokens
+    uint challengeID;       // ID of canonical challenge
+  }
+
+  // Maps of videoId Hashes to listing data.
+  mapping(bytes32 => Listing) public listings;
+
+  ParatiiToken public token;
+  uint public minDeposit;
+  uint public applyStageLen;
+
+  // ---------------------------------------------------------------------------
+  // CONSTRUCTOR
+  // ------------------
+
+  function TcrPlaceholder(
+    address _tokenAddr,
+    uint _minDeposit,
+    uint _applyStageLen
+  ) {
+    token = ParatiiToken(_tokenAddr);
+    minDeposit = _minDeposit;
+    applyStageLen = _applyStageLen; // 100 blocks.
```

Comment:
This is actually a bit of an issue (that I'd liek to ignore for now). We do "instant mining" on our side chain, so there is no predictable block time

Reply:
actually we could use `block.timestamp` instead of `block.number` like in the reference implementation. it wasn't working so great with testing. but i'll fix it 👍